stellargraph/connector/neo4j/graph.py

```diff
@@ -36,18 +41,72 @@ class Neo4jStellarGraph:
 
     Args:
         graph (py2neo.Graph): a py2neo.Graph connected to a Neo4j graph database.
+        node_label (str, optional): Common label for all nodes in the graph, if such label exists.
+            Providing this is useful if there are any indexes created on this label (e.g. on node IDs),
+            as it will improve performance of queries.
         is_directed (bool, optional): If True, the data represents a
             directed multigraph, otherwise an undirected multigraph.
     """
 
-    def __init__(self, graph_db, is_directed=False):
+    def __init__(self, graph_db, node_label=None, is_directed=False):
+
+        try:
+            import py2neo
+        except ModuleNotFoundError as e:
+            raise ModuleNotFoundError(
+                f"{e.msg}. StellarGraph can only connect to Neo4j using the 'py2neo' module; please install it",
+                name=e.name,
+                path=e.path,
+            ) from None
+
+        # check for node ID constraint
+        warn_msg = "Queries may be slow to run and may experience unexpected behaviour if there are duplicate IDs. Please consider creating a uniqueness constraint on a label applied to all nodes in the graph."
+        if node_label is None:
+            warnings.warn(
+                "node_label: No node label provided. " + warn_msg,
+                RuntimeWarning,
+                stacklevel=2,
+            )
+        else:
+            constraint_query = """
+                CALL db.constraints
+                """
+            constraint_regex = re.compile(
+                f"^CONSTRAINT ON \( \w+:{node_label} \) ASSERT \(\w+.ID\) IS UNIQUE$"
```

Comment:
The `\`s here might be working strangely because this isn't a raw string; it's somewhat of a coincidence that none of the characters after `\` here are meaningful escapes (like `\n`). Thus, maybe:

```suggestion
                rf"^CONSTRAINT ON \( \w+:{node_label} \) ASSERT \(\w+.ID\) IS UNIQUE$"
```

```suggestion
                f"^CONSTRAINT ON \\( \\w+:{node_label} \\) ASSERT \\(\\w+.ID\\) IS UNIQUE$"
```